### PR TITLE
Feature/bai 836 post api v2 schemas

### PR DESCRIPTION
**Pull request by JR40159**

### CODE REVIEW

File: backend/src/services/v2/schema.ts [hunk 1]
```diff
@@ -1,7 +1,21 @@
+import { MongoServerError } from 'mongodb'
+
 import { testDeploymentSchema, testModelSchema } from '../../../test/testUtils/testModels.js'
 import Schema, { SchemaInterface } from '../../models/v2/Schema.js'
 import { SchemaKindKeys } from '../../types/v2/enums.js'
-import { NotFound } from '../../utils/v2/error.js'
+import { BadReq, NotFound } from '../../utils/v2/error.js'
+
+function isMongoServerError(err: unknown): err is MongoServerError {
```
Comment (a3957273) [inline backend/src/services/v2/schema.ts:8]: Nice, but can we move these error guards somewhere else?  This feels like a generic function.

File: backend/src/utils/__mocks__/user.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { NextFunction, Request, Response } from 'express'
+
+export async function getUser(req: Request, _res: Response, next: NextFunction) {
+  console.log('auto mocked getUser')
```
Comment (a3957273) [inline backend/src/utils/__mocks__/user.ts:4]: Either change these to use the `logger` or remove them?

File: backend/src/utils/__mocks__/user.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  next()
+}
+
+export function ensureUserRole() {
+  return function ensureUserRoleMiddleware(req: Request, _res: Response, next: NextFunction) {
+    console.log('auto mocked ensureUserRole')
```
Comment (a3957273) [inline backend/src/utils/__mocks__/user.ts:10]: Either change these to use the `logger` or remove them?

File: backend/test/routes/schema/__snapshots__/postSchema.spec.ts.snap [hunk 4]
```diff
@@ -0,0 +1,27 @@
+// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html
+
+exports[`routes > schema > postSchema > blah > successfully stores the schema blah 1`] = `
+{
+  "data": {},
```
Comment (a3957273) [inline backend/test/routes/schema/__snapshots__/postSchema.spec.ts.snap:5]: I would expect this to return some data, e.g. the schema object.

File: backend/test/routes/schema/__snapshots__/postSchema.spec.ts.snap [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+{
+  "error": {
+    "message": "You do not have the 'admin' role",
+  },
+}
+`;
+
+exports[`routes > schema > postSchema > successfully stores the schema 1`] = `
+{
+  "data": {},
```
Comment (a3957273) [inline backend/test/routes/schema/__snapshots__/postSchema.spec.ts.snap:19]: Expect this to return a schema.

File: backend/test/routes/schema/__snapshots__/postSchema.spec.ts.snap [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+}
+`;
+
+exports[`routes > schema > postSchema > successfully stores the schema 2`] = `
+{
+  "data": {},
```
Comment (a3957273) [inline backend/test/routes/schema/__snapshots__/postSchema.spec.ts.snap:25]: Expect this to return a schema.

**Review by a3957273 — CHANGES_REQUESTED** [review]

File: backend/src/utils/__mocks__/user.ts (hunk 2 above)
Comment (JR40159) [reply backend/src/utils/__mocks__/user.ts:4]: Oops, added debugging while development. Good spot

**Pull request merged**